### PR TITLE
[DONE] Carrion JSONs

### PR DESCRIPTION
**Pull request by acidia**
This is the JSON data that is utilized by #23833.  Birds and roaches are currently more fleshed out than anything else.  It looks huge because the linter touched everything...

<a href="https://imgur.com/LSMYKMp"></a>

Birds:
-All birds lay eggs, poop (which attracts predators), and most can spawn from wild "bird eggs"
-Chicken lays eggs year round, poops like a factory, rare for wild eggs
-Grouse, eggs in spring/summer, low biosignature to avoid predators
-Duck/Goose, eggs in spring only, no special water tags yet
-Turkey, eggs in spring/summer, slower to age because of large size
-Pheasant, eggs in spring only but large numbers, high dodge fast bird
-Cockatrice, eggs in spring/summer/fall, high dodge, fast, just a mutant bird that has the better traits of the others, only tiny chance of spawn from bird eggs but not wild ones.

Insects and Spiders:
-Roaches have a full life cycle and leave a biosignature that spawns giant spiders (which are actually weaker than two roaches... oh well)
-Plague roaches are roaches that have been eating tainted meat and will avoid combat till they are older, making their infestations a bigger pain.  Lays eggs and leaves biosignatures just like normal roaches
-Ant eggs now rot into ant larva which can grow to ant soldiers after a while
-Didn't change spiders just because I didn't want to create a pile of different species of baby spider so I'll probably do the same thing I did with chicks and make all baby spiders appear the same?
-Locusts spawn on growing plants, aren't huge threats but can leap away or into combat, full life cycle

Mammals:
-Bears have can have 1 cub a year in spring that take 3 seasons to grow
-Cows can have up to 2 calves a year between spring/summer/fall and take 2 seasons to grow.  Manure factories
-Deer can have up to 2 fawns a year between spring/summer and take 1 season to grow
-Pigs can have 3 piglets in Spring and Fall and take 20 days to grow, second best poop factory
-Sheep can have a lamb between spring/summer/fall every 20 days and it takes 20 days to grow
-Biosignatures from all of these large animals can spawn large carnivores in the area

Triffid:
-Triffid sprigs spawn on rotting clean meat and can grow to young triffids then adults over the course of a few seasons
-biollante springs follow the same growth cycle as triffids

Fish:
-Most fish lay a generic roe item in their respective seasons that can be harvested to eat
-Didn't add baby fish because there is little reason to yet, also adding water flags would be a pain

Reptiles:
-Only the giant rattlesnake lays eggs in spring, all the others are too rare or small

Next Up:
[DONE] Filling the recipes with the new eggs and condensing crafting lists (since all different bird eggs are called "bird egg" it'll be easy to prevent the UI from printing that X10)

### CODE REVIEW

File: data/json/items/comestibles/egg.json [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    "type": "COMESTIBLE",
+    "id": "spider_egg",
+    "name": "spider egg",
+    "weight": 220,
+    "color": "white",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "quench": 10,
+    "nutrition": 20,
+    "description": "A fist sized egg from from a giant spider.  Incredibly gross.",
```
Comment (DracoGriffin) [inline data/json/items/comestibles/egg.json]: Recommend correcting to: `A fist-sized egg from a giant spider.  Incredibly gross.`

File: data/json/items/comestibles/egg.json [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    "name": "roach egg",
+    "weight": 60,
+    "color": "white",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "spoils_in": 112,
+    "healthy": -1,
+    "quench": 3,
+    "nutrition": 8,
+    "description": "A fist sized egg from from a giant roach.  Incredibly gross.",
```
Comment (DracoGriffin) [inline data/json/items/comestibles/egg.json]: Recommend correcting to: `A fist-sized egg from a giant roach.  Incredibly gross.`

File: data/json/materials.json [hunk 3]
```diff
@@ -906,5 +882,25 @@
...
+    "ident": "feces",
+    "name": "Feces",
+    "density": 4,
+    "bash_resist": 1,
+    "cut_resist": 1,
+    "acid_resist": 1,
+    "fire_resist": 0,
+    "elec_resist": 1,
+    "chip_resist": 1,
+    "dmg_adj": [ "squished", "squshed", "mashed", "mushed" ],
```
Comment (DracoGriffin) [inline data/json/materials.json]: Although "squshed" is technically a word, it's incredibly rare. Maybe "squashed" ?

**Comment by DracoGriffin** [comment]
Great work. I would've recommended holding off on the linter, it tends to make reviewing much more tedious and messy.

**Comment by smolbird** [comment]
So we're technically adding shit now. I bet this is something all the people joking about "when will they add the need to defecate" never thought about.

**Comment by acidia** [comment]
Still no character or NPC pooping!

**Comment by smolbird** [comment]
> Still no character or NPC pooping!

Here's hoping.

File: data/json/materials.json [hunk 4]
```diff
@@ -906,5 +882,25 @@
       { "fuel": 1, "smoke": 1, "burn": 2 },
       { "fuel": 1, "smoke": 1, "burn": 5 }
     ]
+  },
+  {
+    "type": "material",
+    "ident": "feces",
+    "name": "Feces",
```
Comment (ZhilkinSerg) [inline data/json/materials.json]: @kevingranade: Are you okay with that?

Comment (kevingranade) [reply data/json/materials.json]: Yea, the issue has always been with player elimination being pointless to depict, not that I want to insist shit doesn't exist.

**Pull request merged**